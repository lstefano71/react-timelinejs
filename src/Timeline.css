/* axis */

/*.axis { !* axis labels *!*/
    /*fill: #808080;*/
    /*font-family: sans-serif;*/
    /*font-size: 10px;*/
/*}*/

/*.axis line{ !* axis tick marks *!*/
    /*stroke-width : 1;*/
    /*stroke: grey;*/
    /*shape-rendering: crispEdges;*/
/*}*/

/*.axis path { !* axis line *!*/
    /*stroke-width : 1;*/
    /*stroke: grey;*/
    /*shape-rendering: crispEdges;*/
/*}*/

/* timeline band */

.timeline {
    position: relative;
}

.band { /* band background */
    fill: #FAFAFA;
}

/* labels */

.bandLabel {
    fill: #F0F0F0;
    font: 10px sans-serif;
    font-weight: bold;
}

.bandMinMaxLabel {
    fill: blue;
    font: 10px sans-serif;
    font-weight: bold;
}

.bandMidLabel {
    fill: red;
    font: 10px sans-serif;
    font-style: italic;
    font-weight: bold;
}

/* brush */

.brush .extent {
    stroke: gray;
    fill: blue;
    fill-opacity: .1;
}

.chart {
    fill: #EEEEEE;
}

.interval {
    fill: dodgerblue;
    stroke-width: 6;
    cursor : default;
    pointer-events: true;
}

.instant {
    fill: dodgerblue;
    stroke-width: 6;
    cursor : default;
    /*pointer-events: true;*/
}

.instantLabel {
    fill : blue;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
}

.intervalLabel {
    fill : black;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
}

.item {
    cursor : default;
    pointer-events: auto;
}

/*.svg {*/
    /*border-style: solid;*/
    /*border-width: 1px;*/
    /*border-color: black;*/
    /*background-color: #FFFFFF;*/
/*}*/

.tooltip {
    width: 0px;
    height: 0px;
    position: absolute;
    z-index: 2;
    visibility: hidden;
    color : black;
    cursor:default;
    pointer-events: none;
    display: flex;
}

.tooltip.top-left {

}
.tooltip.top-right {
    flex-direction: row-reverse;
}
.tooltip.bottom-right {
    flex-flow: row-reverse wrap-reverse;
}
.tooltip.bottom-left {
    flex-flow: column-reverse;
}

.tooltip-container {
    shape-rendering: crispEdges;
    padding: 4px;
    min-width: 300px;
    min-height: 100px;
    border: 1px solid;
    background-color: #FFFFEE;
}

.mouseover-line {
    stroke: #0984e3;
}

.mouseover-label {
    fill : white;
    font: 10px sans-serif;
    shape-rendering: crispEdges;
    text-anchor: middle;
}

.mouseover-label-background {
    fill: #0984e3;
}
