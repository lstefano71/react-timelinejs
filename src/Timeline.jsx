import React from 'react';
import timeline from './TimelineModel';
import './Timeline.css';
import * as R from 'ramda';


// const propTypes = {
//     label: PropTypes.string.isRequired,
//     onChange: PropTypes.func.isRequired,
//     styles: PropTypes.object
// };
//
// const defaultProps = {
//     styles: {
//         label: {
//             fontFamily: 'Comic Sans MS',
//             color: 'green'
//         },
//         input: {
//             background: '#ddd',
//             border: '1px solid red'
//         }
//     }
// };

class Timeline extends React.Component {

    componentDidMount() {
        this.key = Math.round(Math.random() * 100000).toFixed(0) + '-' + new Date().getTime();
        this.createTimeline();
    }


    componentDidUpdate() {
    }

    componentWillReceiveProps(nextProps, nextState) {
        if (!R.equals(nextProps.range, this.props.range)){
            this.timelineFn.updateRange(nextProps.range);
        }
        if (!R.equals(nextProps.brushRange, this.props.brushRange)){
            this.timelineFn.updateBrushRange(nextProps.brushRange);
        }
    }

    createTimeline = () => {

        const {data, lines, height, width, label, tooltips, brush, brushRange, onBrush, onMouseover, range} = this.props;
        const config = {
            width,
            height,
            label,
            tooltips,
            brush,
            brushRange,
            range,
            onBrush,
            onMouseover
        };
        this.timelineFn = timeline(this.div, config).create(data, lines);
        this.timelineFn.redraw();
    };

    render() {
        return <div ref={input => this.div = input}/>
    }
}

// Timeline.propTypes = propTypes;
// Timeline.defaultProps = defaultProps;

export default Timeline;
