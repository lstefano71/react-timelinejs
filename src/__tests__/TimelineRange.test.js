import React from 'react';
import Timeline from '../Timeline';
import {mockData, mockLines, mockRange} from './Timeline.tester';

test('render with range', () => {

    const data = mockData({size: 3});
    const lines = mockLines({size: 10});
    const range = mockRange(data);
    const wrapper = mount(<Timeline id={'1'} data={data} range={range} lines={lines}/>);
    expect(html(wrapper)).toMatchSnapshot();

});
