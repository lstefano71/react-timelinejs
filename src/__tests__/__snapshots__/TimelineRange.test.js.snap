// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with range 1`] = `
"
<div>
  <svg class=\\"svg\\"
       id=\\"svg\\"
       width=\\"700\\"
       height=\\"400\\"
  >
    <g transform=\\"translate(0,0)\\">
      <clipPath id=\\"chart-area\\">
        <rect width=\\"700\\"
              height=\\"400\\"
        >
        </rect>
      </clipPath>
      <g class=\\"chart\\"
         clip-path=\\"url(#chart-area)\\"
      >
        <g id=\\"band0\\"
           transform=\\"translate(0,0)\\"
        >
          <rect class=\\"band\\"
                width=\\"700\\"
                height=\\"380\\"
          >
          </rect>
          <path d=\\"M-878.6692595940302 0 L-878.6692595940302 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-814.9444020848346 0 L-814.9444020848346 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-751.2195445756389 0 L-751.2195445756389 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-687.4946870664434 0 L-687.4946870664434 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-623.7698295572478 0 L-623.7698295572478 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-560.0449720480522 0 L-560.0449720480522 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-496.3201145388566 0 L-496.3201145388566 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-432.59525702966096 0 L-432.59525702966096 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-368.87039952046536 0 L-368.87039952046536 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M-305.1455420112698 0 L-305.1455420112698 380\\"
                class=\\"line test\\"
          >
          </path>
          <path d=\\"M0 0 L0 380\\"
                class=\\"mouseover-line\\"
                style=\\"display: none;\\"
          >
          </path>
          <svg y=\\"4\\"
               height=\\"16\\"
               class=\\"part interval\\"
               x=\\"971.4060267733414\\"
               width=\\"78.59399092800777\\"
          >
            <rect width=\\"100%\\"
                  height=\\"100%\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"10%\\"
                  y=\\"0\\"
                  width=\\"20%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"40%\\"
                  y=\\"0\\"
                  width=\\"9.999999999999998%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"55.00000000000001%\\"
                  y=\\"0\\"
                  width=\\"19.999999999999996%\\"
                  height=\\"16\\"
            >
            </rect>
            <text class=\\"intervalLabel\\"
                  x=\\"1\\"
                  y=\\"10\\"
            >
              event 2
            </text>
          </svg>
          <svg y=\\"4\\"
               height=\\"16\\"
               class=\\"part interval\\"
               x=\\"-259.8693670766232\\"
               width=\\"48.85572409038329\\"
          >
            <rect width=\\"100%\\"
                  height=\\"100%\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"10%\\"
                  y=\\"0\\"
                  width=\\"20%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"40%\\"
                  y=\\"0\\"
                  width=\\"9.999999999999998%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"55.00000000000001%\\"
                  y=\\"0\\"
                  width=\\"19.999999999999996%\\"
                  height=\\"16\\"
            >
            </rect>
            <text class=\\"intervalLabel\\"
                  x=\\"1\\"
                  y=\\"10\\"
            >
              event 0
            </text>
          </svg>
          <svg y=\\"4\\"
               height=\\"16\\"
               class=\\"part interval\\"
               x=\\"-350\\"
               width=\\"37.172833547030734\\"
          >
            <rect width=\\"100%\\"
                  height=\\"100%\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"10%\\"
                  y=\\"0\\"
                  width=\\"20%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"40%\\"
                  y=\\"0\\"
                  width=\\"9.999999999999998%\\"
                  height=\\"16\\"
            >
            </rect>
            <rect class=\\"step custom_1\\"
                  x=\\"55.00000000000001%\\"
                  y=\\"0\\"
                  width=\\"19.999999999999996%\\"
                  height=\\"16\\"
            >
            </rect>
            <text class=\\"intervalLabel\\"
                  x=\\"1\\"
                  y=\\"10\\"
            >
              event 1
            </text>
          </svg>
        </g>
        <g class=\\"axis\\"
           transform=\\"translate(0,380)\\"
           fill=\\"none\\"
           font-size=\\"10\\"
           font-family=\\"sans-serif\\"
           text-anchor=\\"middle\\"
        >
          <path class=\\"domain\\"
                stroke=\\"#000\\"
                d=\\"M0.5,6V0.5H700.5V6\\"
          >
          </path>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(13.978745534708198,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              02 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(77.7036030439038,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              03 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(141.4284605530994,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              04 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(205.15331806229497,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              05 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(268.8781755714906,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              06 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(332.6030330806862,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              07 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(396.32789058988175,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              08 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(460.0527480990774,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              09 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(523.777605608273,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              10 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(587.5024631174686,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              11 PM
            </text>
          </g>
          <g class=\\"tick\\"
             opacity=\\"1\\"
             transform=\\"translate(651.2273206266642,0)\\"
          >
            <line stroke=\\"#000\\"
                  y2=\\"6\\"
            >
            </line>
            <text fill=\\"#000\\"
                  y=\\"9\\"
                  dy=\\"0.71em\\"
            >
              Jul 15
            </text>
          </g>
        </g>
      </g>
    </g>
  </svg>
  <div class=\\"tooltip\\"
       style=\\"visibility: visible;\\"
  >
  </div>
</div>
"
`;
